pkg/app/AppService.go

```diff
@@ -2195,47 +2281,142 @@ func (impl *AppServiceImpl) getConfigMapAndSecretJsonV2(appId int, envId int, pi
 	var configMapJsonEnv string
 	var secretDataJsonEnv string
 	var err error
+	var scope resourceQualifiers.Scope
+	var entitiesForCM []repository6.Entity
+	var entitiesForCS []repository6.Entity
+	var entityToVariablesCM map[repository6.Entity][]string
+	var entityToVariablesCS map[repository6.Entity][]string
+	var resolvedTemplateCM string
+	var variableMapCM map[string]string
+	var resolvedTemplateCS string
+	var variableMapCS map[string]string
 	//var configMapJsonPipeline string
 	//var secretDataJsonPipeline string
 
 	merged := []byte("{}")
-	if deploymentWithConfig == bean.DEPLOYMENT_CONFIG_TYPE_LAST_SAVED {
-		configMapA, err := impl.configMapRepository.GetByAppIdAppLevel(appId)
+	if cMCSJson.DeploymentWithConfig == bean.DEPLOYMENT_CONFIG_TYPE_LAST_SAVED {
+		//cMCSJson.envOverride = &chartConfig.EnvConfigOverride{}
+
+		//VARIABLE different cases for variable resolution
+		env, err := impl.envRepository.FindById(cMCSJson.EnvId)
+		if err != nil {
+			impl.logger.Errorw("unable to find env", "err", err)
+			return nil, err
+		}
+		scope = resourceQualifiers.Scope{
+			AppId:     cMCSJson.AppId,
+			EnvId:     cMCSJson.EnvId,
+			ClusterId: env.ClusterId,
+			SystemMetadata: &resourceQualifiers.SystemMetadata{
+				EnvironmentName: env.Name,
+				ClusterName:     env.Cluster.ClusterName,
+				Namespace:       env.Namespace,
+				ImageTag:        cMCSJson.ImageTag,
+				AppName:         cMCSJson.AppName,
+			},
+		}
+		configMapA, err := impl.configMapRepository.GetByAppIdAppLevel(cMCSJson.AppId)
 		if err != nil && pg.ErrNoRows != err {
 			return []byte("{}"), err
 		}
 		if configMapA != nil && configMapA.Id > 0 {
 			configMapJsonApp = configMapA.ConfigMapData
 			secretDataJsonApp = configMapA.SecretData
 		}
-		configMapE, err := impl.configMapRepository.GetByAppIdAndEnvIdEnvLevel(appId, envId)
+
+		configMapE, err := impl.configMapRepository.GetByAppIdAndEnvIdEnvLevel(cMCSJson.AppId, cMCSJson.EnvId)
 		if err != nil && pg.ErrNoRows != err {
 			return []byte("{}"), err
 		}
 		if configMapE != nil && configMapE.Id > 0 {
 			configMapJsonEnv = configMapE.ConfigMapData
 			secretDataJsonEnv = configMapE.SecretData
 		}
-	} else if deploymentWithConfig == bean.DEPLOYMENT_CONFIG_TYPE_SPECIFIC_TRIGGER {
+		entitiesForCM = []repository6.Entity{
+			{
+				EntityType: repository6.ConfigMapAppLevel,
+				EntityId:   configMapA.Id,
+			},
+			{
+				EntityType: repository6.ConfigMapEnvLevel,
+				EntityId:   configMapE.Id,
+			},
+		}
+		entitiesForCS = []repository6.Entity{
+			{
+				EntityType: repository6.SecretAppLevel,
+				EntityId:   configMapA.Id,
+			},
+			{
+				EntityType: repository6.SecretEnvLevel,
+				EntityId:   configMapE.Id,
+			},
+		}
+
+		entityToVariablesCM, err = impl.GetEntityToVariableMapping(entitiesForCM)
+		if err != nil {
+			return nil, err
+		}
+		entityToVariablesCS, err = impl.GetEntityToVariableMapping(entitiesForCS)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
why not a single call?

Reply:
Also why can't we move the GetEntityToVariableMapping call to the place where we are parsing